api/routes.py

```diff
@@ -155,17 +155,19 @@ def submit_manifest_route(schema_url):
 
     manifest_record_type = connexion.request.args["manifest_record_type"]
 
+    restrict_rules = connexion.request.args["restrict_rules"]
+
     metadata_model = initalize_metadata_model(schema_url)
 
     input_token = connexion.request.args["input_token"]
 
     if data_type == 'None':
         manifest_id = metadata_model.submit_metadata_manifest(
-            manifest_path=temp_path, dataset_id=dataset_id, validate_component=None, input_token=input_token
+            manifest_path=temp_path, dataset_id=dataset_id, validate_component=None, input_token=input_token, manifest_record_type = manifest_record_type, restrict_rules = restrict_rules
```

Comment:
nitpick:

In the if/else block, just set the variables validate_component to its respective value (i.e. set it to None if data_type is None, else set it to data_type); then call the function 

metadata_model.submit_metadata_manifest(...) after the if/else statement block.

Reply:
Thanks for pointing it out Milen! I think we initially designed it this way just in case someone enters the string "None" for dataset_id in the UI. If we do something like below, the code will break when someone enters the string "None": 
```
def submit_manifest_route(schema_url, data_type=None):
    # call config_handler()
    config_handler()

    # Get path to temp file where manifest file contents will be saved
    temp_path = csv_path_handler()

    dataset_id = connexion.request.args["dataset_id"]

    manifest_record_type = connexion.request.args["manifest_record_type"]

    restrict_rules = connexion.request.args["restrict_rules"]

    metadata_model = initalize_metadata_model(schema_url)

    input_token = connexion.request.args["input_token"]

    manifest_id = metadata_model.submit_metadata_manifest(
            manifest_path=temp_path, dataset_id=dataset_id, validate_component=data_type, input_token=input_token, manifest_record_type = manifest_record_type, restrict_rules = restrict_rules
        )
```